packages/adapter/src/lib/adapter/adapter.ts

```diff
@@ -2434,8 +2434,24 @@ export class AdapterClass extends EventEmitter {
     }
 
     private async _updateConfig(options: InternalUpdateConfigOptions): ioBroker.SetObjectPromise {
+        const { newConfig } = options;
+
         // merge the old and new configuration
-        const _config = Object.assign({}, this.config, options.newConfig);
+        const config: Record<string, unknown> = { ...this.config, ...newConfig };
```

Comment:
why we are using this.config when we read the real 8current!) object some lines below anyway?

this.config can be outdated because of async/onchange triggering and strange effects - and also in this case (/if we would use it) we should also set this.config to the new value and not rely on an async updating :-)

Said that: move the logic down some lines below the "getForeignOnjectAsync" calöll and ideally use config from there directly and ignore this.config.